python/cudf/dataframe/index.py

```diff
@@ -537,6 +561,26 @@ def find_label_range(self, first, last):
             end += 1
         return begin, end
 
+    @property
+    def is_unique(self):
+        return len(set(self._values)) == len(self._values)
+
+    @property
+    def is_monotonic_increasing(self):
+        for val in zip(self._values[1:], self._values[:-1]):
+            if val[0] < val[1]:
```

Comment:
Your idea here is great in using slicing to do a stencil comparison, but we need to rework this a bit to have it run nicely on the device. Instead of zipping them and iterating we can just run a binaryop and then use `all()` to ensure it's all true. This would look like:
```
return (self._values.iloc[1:] >= self._values[:-1]).all()
```

Reply:
Also, we should memoize these instead of recomputing them every time 😄

Reply:
I added some manual memoization - Perhaps I can/should just add a `memoize_property` function to `cudf/utils/utils.py` (and just use a `@memoize_property` decorator throughout)?